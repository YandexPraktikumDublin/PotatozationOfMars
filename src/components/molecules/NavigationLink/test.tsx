import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import { NavigationLink } from '.'

describe('<NavigationLink />', () => {
  const title = 'Test title'
  const href = '/test'
  const imageSrc = 'image.jpg'

  it('should renders correct <NavigationLink />', () => {
    const wrapper = shallow(
      <NavigationLink title={title} href={href} imageSrc={imageSrc} />
    )

    expect(wrapper.prop('to')).toMatch(href)
    expect(wrapper.prop('title')).toMatch(title)
    expect(wrapper.find('img').prop('src')).toMatch(imageSrc)
    expect(wrapper.find('img').prop('alt')).toMatch(title)

    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
