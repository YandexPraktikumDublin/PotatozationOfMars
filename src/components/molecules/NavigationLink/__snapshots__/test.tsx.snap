// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationLink /> should renders correct <NavigationLink /> 1`] = `
<Link
  title="Test title"
  to="/test"
>
  <img
    alt="Test title"
    height="25"
    src="image.jpg"
    width="25"
  />
</Link>
`;
